src/sys.rs

```diff
@@ -455,10 +455,39 @@ mod inner {
             call!(SystemTimeToTzSpecificLocalTime(0 as *const _,
                                                   &mut utc, &mut local));
             system_time_to_tm(&local, tm);
-
+            tm.tm_utcoff = 0;
+            tm.tm_isdst = -1;
+            
             let mut tz = mem::zeroed();
-            GetTimeZoneInformation(&mut tz);
-            tm.tm_utcoff = -tz.Bias * 60;
+            let this_year = 1900 + tm.tm_year as u16;
+            let get_tz_info = GetTimeZoneInformationForYear(this_year, 0 as *mut _, &mut tz) != 0;
+
+            if get_tz_info {
+                tm.tm_isdst = 0;
+                tm.tm_utcoff -= tz.Bias * 60;
+
+                if tz.StandardDate.wMonth != 0 && tz.DaylightDate.wMonth != 0 {
```

Comment:
A few thoughts here:

* Could you add a link here to the MSDN page? Also, could you add a comment indicating that this is basically just trying to figure out if `tm` is within the range of DST and if so it uses the `DaylightBias` (otherwise `StandardBias` is used).
* The comparison operator may be relatively expensive (calls a few extra function calls) and deals with a lot of data, perhaps a more focused comparison could be used? (e.g. one that's entirely contained to just this module)
* It's a little confusing that `tm_utcoff` is set multiple times in this function, could the assignment to `tz.Bias * 60` happen in a fall-through branch that's only taken if `StandardBias` or `DaylightBias` isn't used?